server.go

```diff
@@ -465,12 +516,17 @@ func (srv *Server) serveTCP(l net.Listener) error {
 			}
 			return err
 		}
-		go srv.serveTCPConn(handler, rw)
+		w := &response{tsigSecret: srv.TsigSecret, tcp: rw}
+		select {
+		case srv.queueTCPConn <- w:
+		default:
+			srv.spawnWorkerTCP()
```

Comment:
This will lead to unbounded goroutine growth. Maybe that’s not a problem in reality, but even a small burst of requests would keep a large number of goroutines forever.

Reply:
BTW we have unbounded goroutine growth already in current implementation.

Reply:
@UladzimirTrehubenka Not strictly true, currently the goroutines will be garbage collected once they've served the request. These ones will be kept around forever.

Consider an constant load of 1req/s with a sudden burst of 100req/s for 1s:
- Currently one routine will be created per request leading to a peak of 100 goroutines that *are then garbage collected*.
- This pull request will cause 100 goroutines to be active *forever*.

The above would be particularly bad if a server was swamped with a very large number of bogus requests as in a DDOS situation, because the memory load can't be reclaimed.